### DISCUSSION TITLE
Generate TinyProfiler tables every X timesteps

### DISCUSSION THREAD

**Discussion by psharda** [AMReX-Codes/amrex#3560] [General]
Hi, we are developing an AMReX based hydro code called Quokka for astrophysical simulations (https://github.com/quokka-astro/quokka). We are wondering if it is possible to have a feature that lets the user print the TinyProfiler tables every X timesteps, so that the user can monitor the time/memory consumption of different modules _in real time_. Currently, these tables are only printed at the end of the simulation (e.g. of such a table below):

```
TinyProfiler total time across processes [min...avg...max]: 0.0938 ... 0.0938 ... 0.0938

---------------------------------------------------------------------------------------------------
Name                                                NCalls  Excl. Min  Excl. Avg  Excl. Max   Max %
---------------------------------------------------------------------------------------------------
AMRSimulation::MakeNewLevelFromScratch()                 1    0.02417    0.02417    0.02417  25.77%
RadhydroSimulation::computeHydroFluxes()                 2    0.01504    0.01504    0.01504  16.04%
MLPoisson::Fsmooth()                                   160    0.01035    0.01035    0.01035  11.03%
VisMF::Write(FabArray)                                   4   0.008756   0.008756   0.008756   9.34%
PhysBCFunct::()                                          5   0.004553   0.004553   0.004553   4.85%
MLCellLinOp::applyBC()                                 254   0.003558   0.003558   0.003558   3.79%

```

Such a feature would enable us to track the efficiency and speed of different modules more effectively, and optimize our setup as the simulation progresses.

**Comment by WeiqunZhang** [comment]
Could you try this?
```
#if defined(AMREX_TINY_PROFILING)
amrex::TinyProfiler::Finalize(true); // Here true means flushing
#endif
```

**Reply by psharda** [reply]
Yes, this worked, and now the tables are printed after every timestep. Just need to add a runtime parameter that prints them every X steps.
